src/textbooks/textbook-form/TextbookForm.jsx

```diff
@@ -0,0 +1,192 @@
+      }),
+    ).min(1),
+  });
+
+  return (
+    <div className="textbook-form">
+      <Formik
+        initialValues={initialFormValues}
+        validationSchema={textbookFormValidationSchema}
+        onSubmit={onSubmit}
+        validateOnBlur
+        validateOnMount
+      >
+        {({
+          values, handleSubmit, isValid, dirty, setFieldValue,
+        }) => (
+          <>
+            <Form.Group size="sm" className="form-field">
+              <Form.Label size="sm" className="font-weight-bold form-main-label">
+                {intl.formatMessage(messages.tabTitleLabel)} *
+              </Form.Label>
+              <FormikControl
+                name="tab_title"
+                value={values.tab_title}
+                placeholder={intl.formatMessage(messages.tabTitlePlaceholder)}
+              />
+              <Form.Control.Feedback className="form-helper-text">
+                {intl.formatMessage(messages.tabTitleHelperText)}
+              </Form.Control.Feedback>
+            </Form.Group>
+            <FieldArray
+              name="chapters"
+              render={(arrayHelpers) => (
+                <>
+                  {Boolean(values?.chapters.length) && values.chapters.map(({ title, url }, index) => (
+                    <div className="form-chapters-fields" data-testid="form-chapters-fields">
+                      <Form.Group size="sm" className="form-field">
+                        <Form.Label size="sm" className="form-label font-weight-bold required">
+                          {intl.formatMessage(messages.chapterTitleLabel)} *
+                        </Form.Label>
+                        <FormikControl
+                          name={`chapters[${index}].title`}
+                          value={title}
+                          placeholder={intl.formatMessage(messages.chapterTitlePlaceholder, { value: index + 1 })}
+                        />
+                        <Form.Control.Feedback className="form-helper-text">
+                          {intl.formatMessage(messages.chapterTitleHelperText)}
+                        </Form.Control.Feedback>
+                      </Form.Group>
+                      <Form.Group size="sm" className="form-field">
+                        <div className="d-flex align-items-center mb-1">
+                          <Form.Label size="sm" className="font-weight-bold mb-0">
+                            {intl.formatMessage(messages.chapterUrlLabel)} *
+                          </Form.Label>
+                          <IconButtonWithTooltip
+                            size="sm"
+                            variant="primary"
+                            className="ml-auto field-icon-button"
+                            tooltipContent="Upload"
+                            src={UploadIcon}
+                            iconAs={Icon}
+                            data-testid="chapter-upload-button"
+                            onClick={() => {
```

Comment:
[optional]: Now the component looks quite voluminous. I have a few suggestions to improve the code:
1. Try to place similar callbacks in separate functions above (with a suitable name describing what exactly they do)
2. Break this `TextbookForm` component into logical smaller components

Think about it, maybe something will interest you. My suggestions are solely at your discretion 💯

Reply:
1. Created onUploadButtonClick - fixed
2. Break this TextbookForm component into logical smaller components - I don't think it's necessary, the component isn't too big